modules/media/ajax/FileUpload.php

```diff
@@ -187,34 +187,36 @@ function uploadFile()
 function getUploadFields()
 {
 
-    $db =& Database::singleton();
-
-    $instruments = $db->pselect(
-        "SELECT Test_name FROM test_names ORDER BY Test_name",
-        []
-    );
-    $candidates  = $db->pselect(
-        "SELECT CandID, PSCID FROM candidate ORDER BY PSCID",
-        []
-    );
-
-    $instrumentsList = toSelect($instruments, "Test_name", null);
-    $candidatesList  = toSelect($candidates, "PSCID", null);
-    $candIdList      = toSelect($candidates, "CandID", "PSCID");
-    $visitList       = Utility::getVisitList();
-    $siteList        = Utility::getSiteList(false);
-    $languageList    = Utility::getLanguageList();
+    $db   =& Database::singleton();
+    $user =& User::singleton();
 
+    $recordsQuery = "SELECT c.PSCID, c.CandID, s.CenterID, s.Visit_label, ".
+                    "f.Test_name ".
+                    "FROM candidate c ".
+                    "LEFT JOIN session s USING(CandID) ".
+                    "LEFT JOIN flag f ON (s.ID=f.SessionID) ".
+                    "WHERE c.PSCID NOT LIKE 'scanner%' ".
+                    "AND f.Test_name NOT LIKE 'DDE_%' ";
+
+    $siteIDs = $user->getCenterIDs();
+    if (!$user->hasPermission('access_all_profiles')) {
+        $recordsQuery .= "AND s.CenterID IN(".implode(",", $siteIDs).") ";
```

Comment:
I think this is safe since $siteIDs comes from `$user->getCenterIDs();` above, but we usually use `FIND_IN_SET` so that we can use prepared statements.

For instance, see: https://github.com/aces/Loris/blob/aae05e7fbdd537f8535838377e80469b86ef562a/modules/data_team_helper/php/data_team_helper.class.inc#L247-L248 (I'm not sure if that's the best example, it's just the first one that showed up when searching on GitHub

Reply:
this is actually good to know. didn't realize `IN` was not usable in prepared statements

Reply:
@driusan Made changes as suggested.